client/src/modules/stock/exit/exit.js

```diff
@@ -299,6 +316,46 @@ function StockExitController(
       .catch(Notify.handleError);
   }
 
+  // on lot select
+  function onLotSelect(row) {
+    if (!row.lot || !row.lot.uuid) { return; }
+
+    checkValidity();
+
+    refreshSelectedLotsList();
+  }
+
+  // update the list of selected lots
+  function refreshSelectedLotsList() {
+    vm.selectedLots = vm.stockForm.store.data
+      .filter(item => item.lot && item.lot.uuid)
+      .map(item => item.lot.uuid);
+  }
+
+  // detect the presence of duplicated lots
+  function hasDuplicatedLots() {
+    refreshSelectedLotsList();
+
+    let doublonDetected = false;
+    for (let i = 0; i < vm.selectedLots.length; i++) {
```

Comment:
It looks like `vm.selectedLots` is an array of `uuid`s.  That means, if you check the first value, you never have to check it again.  You can increase the performance by making `j` start at `i + 1`.  Like this:

```js
for (let i = 0; i < vm.selectedLots.length; i++) {
 let found = 0;
  for (let j = i + 1; j < vm.selectedLots.length; j++) {
    if (lot === vm.selectedLots[j]) {
      found++;
      if (found > 1) { break; }
    }
   // etc..
  }
}
```

I still wish we had solved this [with a datastructure though](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2544#issuecomment-366499969).  It would have been easier to read and more performant.